app/jobs/regular/sync_topic_user_bookmarked.rb

```diff
@@ -5,35 +5,37 @@ class SyncTopicUserBookmarked < ::Jobs::Base
     def execute(args = {})
       raise Discourse::InvalidParameters.new(:topic_id) unless args[:topic_id].present?
 
-      DB.exec(<<~SQL, topic_id: args[:topic_id])
-        SELECT bookmarks.user_id, COUNT(*)
-        INTO TEMP TABLE tmp_sync_topic_user_bookmarks
+      bookmarks_exist = DB.exec(<<~SQL, topic_id: args[:topic_id]) > 0
```

Comment:
Since this is a query for a value now, this should be `DB.query_single` . IIRC the return value of `exec` is based on success of the query, not the actually return value.